### PR TITLE
Shellcheck bash completion

### PR DESCRIPTION
**Pull request by jphuynh**
Breaking down the work done for #32678 in more manageable small PRs.

This one includes bash completion and add shellcheck to the validation process of the build.

cc: @albers and @dnephin

### CODE REVIEW

File: contrib/completion/bash/docker [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env bash
+# shellcheck disable=SC2119,SC2120,SC2155
```
Comment (AkihiroSuda) [inline contrib/completion/bash/docker]: could you add descriptions about these disabled checks?

**Comment by AkihiroSuda** [comment]
CI failing (because shellcheck working :smile:)

**Comment by jphuynh** [comment]
Looks like the global disable rule doesn't work because of the version of shellcheck packaged for debian jessie.

```
root@662ff4830564:/# shellcheck --version
ShellCheck - shell script analysis tool
version: 0.3.4
license: GNU Affero General Public License, version 3
website: http://www.shellcheck.net
```

I didn't spot the errors with my installation of shellcheck running version 0.4.6.

File: Dockerfile [hunk 2]
```diff
@@ -206,6 +208,14 @@ RUN git clone https://github.com/go-swagger/go-swagger.git /go/src/github.com/go
 	&& (cd /go/src/github.com/go-swagger/go-swagger && git checkout -q $GO_SWAGGER_COMMIT) \
 	&& go install -v github.com/go-swagger/go-swagger/cmd/swagger
 
+# Install shellcheck from source to get a more recent version on debian jessie
+ENV SHELLCHECK_VERSION v0.4.6
+RUN set -x \
+	&& git clone --branch "$SHELLCHECK_VERSION" https://github.com/koalaman/shellcheck.git /usr/local/shellcheck \
+	&& cd /usr/local/shellcheck \
+	&& export LANG="C.UTF-8" && cabal update && cabal install
```
Comment (albers) [inline Dockerfile]: Maybe changing `LANG` here globally might cause some trouble in later steps.
Does 
```bash
&& LANG="C.UTF-8" cabal update
&& LANG="C.UTF-8" cabal install
```
do the job as well?

File: Dockerfile.aarch64 [hunk 3]
```diff
@@ -162,6 +164,16 @@ RUN git clone https://github.com/go-swagger/go-swagger.git /go/src/github.com/go
 	&& (cd /go/src/github.com/go-swagger/go-swagger && git checkout -q $GO_SWAGGER_COMMIT) \
 	&& go install -v github.com/go-swagger/go-swagger/cmd/swagger
 
+# Install shellcheck from source to get a more recent version on debian jessie
+ENV SHELLCHECK_VERSION v0.4.6
+# Required by cabal (See: https://github.com/haskell/cabal/issues/1883)
+ENV LANG "en_US.UTF-8"
```
Comment (albers) [inline Dockerfile.aarch64]: same as above, can we avoid the `ENV` step here?

Comment (jphuynh) [inline Dockerfile.aarch64]: Yes sorry the second file hasn't been properly updated.

File: Dockerfile (hunk 2 above)
Comment (jphuynh) [inline Dockerfile]: That works fine I just tested it. Thanks.

**Comment by albers** [comment]
@jphuynh Thanks for creating this better reviewable chung of work!

The completions are going to be moved to the `docker/cli` repo where the `docker` command already lives. I'd like to wait for the transition to be completed. @tiborvass can you tell when this will be done?

Probably this PR can be transferred in this proces as well.

**Comment by jphuynh** [comment]
Sure that's fine. I don't know how you move PRs around between projects but if it is transparent it's fine by me. :)

Let me know if you need me to do anything when that part of code is moved to `docker/cli`

**Comment by cpuguy83** [comment]
Yep, completions have all moved to github.com/docker/cli.
Unfortunately there is not a way to move a PR from one project to another, it will just have to be a new PR on the new repo.

I'm going to close this because it can't be merged here.
Let us know if you'd like someone to help get this onto the new repo.
Thanks!

**Pull request closed without merging**